recipes/full_finetune_single_device.py

```diff
@@ -310,6 +317,11 @@ def train(self) -> None:
         The core training loop. Supports training on subsets of the dataset using the
         ``max_steps_per_epoch``.
         """
+
+        if self._model_compile:
+            log.info(
+                "NOTE: torch.compile is enabled and model is compiled in first forward. Expect a relatively slow first iteration."
```

Comment:
Roughly how much is the upfront cost here? (I know you mentioned you'll give more details soon so fine to wait until then)

Reply:
Seeing 100-300s depending on the recipe. For example, QLoRA seems to need longer to compile, possibly due to more compiler passes for fusing quantization kernels (unsure)